tests/Feature/ContactTest.php

```diff
@@ -0,0 +1,281 @@
+
+        $body = $this->faker->paragraph();
+
+        $this->post('/people/' . $contact->id . '/note/save', [
+            'body' => $body
+        ]);
+
+        $this->assertDatabaseHas('notes', [
+            'contact_id' => $contact->id,
+            'account_id' => $user->account_id,
+            'body' => $body
+        ]);
+
+    }
+
+    public function test_user_can_add_activity_to_contact()
+    {
+        extract($this->createUserAndContact());
+
+        $activity = [
+            'summary' => $this->faker->sentence('5'),
+            'specific_date' => $this->faker->date('Y-m-d'),
+            'comment' => $this->faker->paragraph()
+        ];
+
+        $this->post(
+            '/people/' . $contact->id . '/activities/store',
+                $activity
+        );
+
+        $activity['contact_id'] = $contact->id;
+        $activity['account_id'] = $user->account_id;
+
+        unset($activity['specific_date']);
+
+        // The name of the form element is different to the name of the
+        // database table so this must be changed.
+        $this->changeArrayKey('comment', 'description', $activity);
+
+        $this->assertDatabaseHas('activities', $activity);
+    }
+
+    public function test_user_can_be_reminded_about_an_event_once()
+    {
+        extract($this->createUserAndContact());
+
+        $reminder = [
+            'reminder' => $this->faker->sentence('5'),
+            'reminderNextExpectedDate' => $this->faker->dateTimeBetween('now', '+2 years'),
+            'frequencyType' => 'once',
+            'comment' => $this->faker->sentence()
+        ];
+
+        $this->post(
+            '/people/' . $contact->id . '/reminders/store',
+            $reminder
+        );
+
+        $reminder['contact_id'] = $contact->id;
+        $reminder['account_id'] = $user->account_id;
+
+        $this->changeArrayKey('reminder', 'title', $reminder);
+        $this->changeArrayKey('comment', 'description', $reminder);
+        $this->changeArrayKey('frequencyType', 'frequency_type', $reminder);
+        $reminder['frequency_type'] = 'one_time';
+        unset($reminder['reminderNextExpectedDate']);
+
+        $this->assertDatabaseHas('reminders', $reminder);
+
+    }
+
+    public function test_user_can_add_a_task_to_a_contact()
+    {
+        extract($this->createUserAndContact());
+
+        $task = [
+            'title' => $this->faker->sentence(),
+            'comment' => $this->faker->sentence(3)
+        ];
+
+        $this->post(
+            '/people/' . $contact->id . '/tasks/store',
+            $task
+        );
+
+        $task['contact_id'] = $contact->id;
+        $task['account_id'] = $user->account_id;
+
+        // Change keys to match database column names
+        $this->changeArrayKey('comment', 'description', $task);
+
+        $this->assertDatabaseHas('tasks', $task);
+    }
+
+    /*
+     * Waiting until encryption has been removed from gifts before this test
```

Comment:
Encryption has been removed from the gifts table https://github.com/monicahq/monica/pull/249